Services/CO.CDP.EntityVerification/Events/OrganisationRegisteredEventHandler.cs

```diff
@@ -0,0 +1,37 @@
+using CO.CDP.EntityVerification.Model;
+using CO.CDP.EntityVerification.Persistence;
+using CO.CDP.EntityVerification.Services;
+
+namespace CO.CDP.EntityVerification.Events;
+
+public class OrganisationRegisteredEventHandler : IEventHandler
+{
+    private readonly IPponService _pponService;
+    private readonly IServiceProvider _serviceProvider;
+    private readonly IServiceProviderWrapper _spWrapper;
+
+    public OrganisationRegisteredEventHandler(IPponService pponService,
+        IServiceProvider serviceProvider,
+        IServiceProviderWrapper spWrapper)
+    {
+        _pponService = pponService;
+        _serviceProvider = serviceProvider;
+        _spWrapper = spWrapper;
+    }
+
+    public void Action(EvMessage msg)
+    {
+        OrganisationRegisteredMessage newOrg = (OrganisationRegisteredMessage)msg;
+        var pponid = _pponService.GeneratePponId(newOrg.Scheme, newOrg.GovIdentifier);
+
+        Ppon newIdentifier = new() { PponId  = pponid };
+
+        using (var scope = _serviceProvider.CreateScope())
+        {
+            var dbContext = _spWrapper.GetRequiredService(scope.ServiceProvider);
+            DatabasePponRepository db = new DatabasePponRepository(dbContext);
```

Comment:
Any reason, to do it like this, rather than passing the Ppon repository in the constructor? 

Having a concrete implementation instantiated here defeats the purpose of having the interface.

Reply:
True, updated to inject the DatabasePponRepository.